src/bundle/Resources/public/js/scripts/udw/select.location.js

```diff
@@ -0,0 +1,88 @@
+(function (global, doc, React, ReactDOM) {
+    const btns = doc.querySelectorAll('.btn--udw-select-location');
+    const udwContainer = doc.getElementById('react-udw');
+    const token = doc.querySelector('meta[name="CSRF-Token"]').content;
+    const siteaccess = doc.querySelector('meta[name="SiteAccess"]').content;
+
+    const findLocationsByPathString = (pathString, callback) => {
+        const body = JSON.stringify({
+            ViewInput: {
+                identifier: `udw-locations-by-path-string-${pathString.replace(/\//g, '-')}`,
+                public: false,
+                LocationQuery: {
+                    Criteria: {},
+                    FacetBuilders: {},
+                    SortClauses: { SectionIdentifier: 'ascending' },
+                    Filter: { LocationIdCriterion: pathString.replace(/\//g, ',') },
+                    limit: 50,
+                    offset: 0,
+                },
+            },
+        });
+
+        const request = new Request('/api/ezp/v2/views', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.ez.api.View+json; version=1.1',
+                'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+                'X-Requested-With' : 'XMLHttpRequest',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token
+            },
+            body,
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
+
+        fetch(request)
+            .then(window.eZ.helpers.request.getJsonFromResponse)
+            .then((json) => callback(json))
+            .catch(() => window.eZ.helpers.notification.showErrorNotification('Cannot find children locations by a parent location id'));
+    };
+
+    const removeRootFromPathString = (pathString) => {
+        const pathArray = pathString.split('/').filter((val) => val);
+
+        return pathArray.splice(1, pathArray.length - 1).join('/');
```

Comment:
If you return here an array you will not need to do `replace` but just `join('-')` and second `join()`.
So it's 2 joins vs join, 2 replaces.